gpu4pyscf/nac/mecp.py

```diff
@@ -0,0 +1,243 @@
+                    isinstance(self.solver_wrapper.ci_optimizer.td, tdscf.ris.TDDFT):
+                self.solver_wrapper.ci_optimizer.td.n_occ = None
+                self.solver_wrapper.ci_optimizer.td.n_vir = None
+                self.solver_wrapper.ci_optimizer.td.rest_occ = None
+                self.solver_wrapper.ci_optimizer.td.rest_vir = None
+                self.solver_wrapper.ci_optimizer.td.C_occ_notrunc = None
+                self.solver_wrapper.ci_optimizer.td.C_vir_notrunc = None
+                self.solver_wrapper.ci_optimizer.td.C_occ_Ktrunc = None
+                self.solver_wrapper.ci_optimizer.td.C_vir_Ktrunc = None
+                self.solver_wrapper.ci_optimizer.td.delta_hdiag = None
+                self.solver_wrapper.ci_optimizer.td.hdiag = None
+                self.solver_wrapper.ci_optimizer.td.eri_tag = None
+                self.solver_wrapper.ci_optimizer.td.auxmol_J = None
+                self.solver_wrapper.ci_optimizer.td.auxmol_K = None
+                self.solver_wrapper.ci_optimizer.td.lower_inv_eri2c_J = None
+                self.solver_wrapper.ci_optimizer.td.lower_inv_eri2c_K = None
+                self.solver_wrapper.ci_optimizer.td.RKS = True
+                self.solver_wrapper.ci_optimizer.td.UKS = False
+                self.solver_wrapper.ci_optimizer.td.mo_coeff = cp.asarray(self.solver_wrapper.ci_optimizer.mf.mo_coeff, 
+                    dtype=self.solver_wrapper.ci_optimizer.td.dtype)
+                self.solver_wrapper.ci_optimizer.td.build()
+            return self.kernel(**kwargs)
+
+        def kernel(self, *args, **kwargs):
+            self.solver_wrapper.ci_optimizer.get_eff_energy_and_gradient()
+            return self.solver_wrapper.ci_optimizer._last_energy, self.solver_wrapper.ci_optimizer._last_grad
+
+        def as_scanner(self):
+            """Returns the scanner object."""
+            return self
+
+        def converged(self):
+            td_scanner = self.base
+            return all((td_scanner._scf.converged,
+                        td_scanner.converged[self.states[0]-1],
+                        td_scanner.converged[self.states[1]-1]))
+
+    class _SolverWrapper:
+        def __init__(self, ci_optimizer):
+            self.ci_optimizer = ci_optimizer
+            self.mol = self.ci_optimizer.mol
+            self.verbose = self.ci_optimizer.verbose
+            self.stdout = self.ci_optimizer.stdout
+            
+        def __call__(self, *args, **kwargs):
+            energy, grad = self.ci_optimizer.get_eff_energy_and_gradient()
+            return energy, grad
+
+        def nuc_grad_method(self):
+            return self.ci_optimizer._GradScanner(self)
+
+
+    def get_eff_energy_and_gradient(self):
+        current_geom = self.mol.atom_coords(unit='bohr')
+
+        self.log.info("\n--- CI Optimizer Step (using pyscf.geomopt.geometric_solver) ---")
+
+        self.mf.mol = self.mol
+        self.td.mol = self.mol
+
+        self.mf.kernel()
```

Comment:
These kernel may not be able to correctly apply initial guess from previous calculations